src/Core/Serialization/ProjectLoader.cs

```diff
@@ -202,6 +202,8 @@ public Project LoadProject(ImageLocation projectUri, Project_v5 sp)
             project.MetadataFiles.AddRange(typelibs);
             project.ScriptFiles.AddRange(scripts);
             project.Programs.AddRange(programs);
+
+            this.project.FireScriptEvent(ScriptEvent.OnProgramLoaded);
```

Comment:
Duplicated fire of `OnProgramLoaded` event. It caused regression at https://github.com/uxmal/reko/pull/1098#discussion_r758202324

Reply:
Another one is https://github.com/uxmal/reko/blob/8500b5794287d1f63e228add20422d93a8d8f393/src/Decompiler/Loading/Loader.cs#L108

Reply:
Is it appropriate to say, that the `OnProgramLoaded` event should only be fired if a project has been loaded? If a user opened a `Program` directly, there hasn't been time for the user to add a script to the project, so it makes no sense to fire the event in that case.

Perhaps we should write unit tests to cover for how this is expected to work?

Reply:
> If a user opened a Program directly, there hasn't been time for the user to add a script to the project,

Yes.

>Perhaps we should write unit tests to cover for how this is expected to work?

It would be nice.